### PR TITLE
Feature: pull action

### PR DESCRIPTION
**Pull request by Zorin95670**
Add pull action in git menu.

### CODE REVIEW

File: src/composables/Project.js [hunk 1]
```diff
@@ -280,3 +280,31 @@ export async function createBranchFrom(projectId, newBranchName, branchName, hav
     checkout(projectId, newBranchName);
   }
 }
+
+/**
+ * Make a git pull.
```
Comment (justinehell) [inline src/composables/Project.js:285]: `Update local project with git pull` ?

File: src/components/menu/GitBranchActionMenu.vue [hunk 2]
```diff
@@ -24,6 +24,20 @@
...
       </q-item>
+
+      <template v-if="onLocal">
+        <q-separator/>
+
+        <q-item
+            :data-cy="`git-menu-branch-update-${branchName}`"
+            clickable
+            @click="onUpdate"
+        >
```
Comment (justinehell) [inline src/components/menu/GitBranchActionMenu.vue:35]: wrong indents

File: src/components/menu/GitBranchActionMenu.vue [hunk 3]
```diff
@@ -24,6 +24,20 @@
           {{ $t('actions.git.newBranchFrom', { branch: branchName }) }}
         </q-item-section>
       </q-item>
+
+      <template v-if="onLocal">
```
Comment (justinehell) [inline src/components/menu/GitBranchActionMenu.vue:28]: not convinced by the naming `onLocal`. Maybe something more specific ?

File: cypress/e2e/git/branch_menu_update.feature [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    And I expect "[data-cy=\"git-form\"]" is closed
+    And I expect ".q-notification" not exists
+
+    When I click on "[data-cy=\"git-current-branch\"]"
+    And I click on "[data-cy=\"git-menu-branch-remote-main\"]"
+    And I click on "[data-cy=\"git-menu-branch-checkout-main\"]"
+    Then I expect "[data-cy=\"git-current-branch\"] " is "main"
+    And I click on body to close popup
+
+  Scenario: Update action men should be available only for local branches
```
Comment (justinehell) [inline cypress/e2e/git/branch_menu_update.feature:30]: remove `men` ? I don't understand the scenario x)

**Review by justinehell — CHANGES_REQUESTED** [review]

File: src/composables/Project.js (hunk 1 above)
Comment (Zorin95670) [reply src/composables/Project.js:285]: done

File: src/components/menu/GitBranchActionMenu.vue (hunk 2 above)
Comment (Zorin95670) [reply src/components/menu/GitBranchActionMenu.vue:35]: done

File: src/components/menu/GitBranchActionMenu.vue (hunk 3 above)
Comment (Zorin95670) [reply src/components/menu/GitBranchActionMenu.vue:28]: I take the same name of Branch attributes to be consistent

File: src/composables/Project.js [hunk 5]
```diff
@@ -280,3 +280,31 @@ export async function createBranchFrom(projectId, newBranchName, branchName, hav
     checkout(projectId, newBranchName);
   }
 }
+
+/**
+ * Update local project with git pull.
```
Comment (k-y0u) [inline src/composables/Project.js:285]: update all the project or only the (current) branch

File: src/components/form/GitUpdateForm.vue [hunk 6]
```diff
@@ -0,0 +1,83 @@
+<template>
+  <q-form
+    ref="form"
+    @submit="onSubmit"
+    data-cy="git-update-form"
+    class="q-gutter-md git-update-form"
+  >
+    <div v-html="$t('page.modelizer.git.update.description', { branch: branchName })"></div>
+
```
Comment (k-y0u) [inline src/components/form/GitUpdateForm.vue:9]: remove linebreak

**Review by k-y0u — CHANGES_REQUESTED** [review]

**Review by k-y0u — APPROVED** [review]

**Review by k-y0u — APPROVED** [review]

**Review by justinehell — APPROVED** [review]

**Pull request merged**